src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -541,16 +553,10 @@ static void pf_connectivity_changed_locked(grpc_exec_ctx* exec_ctx, void* arg,
         if (sd == original_sd) {
           grpc_lb_subchannel_list_unref_for_connectivity_watch(
               exec_ctx, sd->subchannel_list, "pf_candidate_shutdown");
-          shutdown_locked(exec_ctx, p,
-                          GRPC_ERROR_CREATE_REFERENCING_FROM_STATIC_STRING(
-                              "Pick first exhausted channels", &error, 1));
+          grpc_lb_policy_try_reresolve(
+              exec_ctx, &p->base, &grpc_lb_pick_first_trace, GRPC_ERROR_NONE);
           return;
         }
-        if (sd->subchannel_list == p->subchannel_list) {
```

Comment:
Line 549 to line 553 is moved up to make sure we set the state even when the whole subchannel_list is invalid.